### ISSUE TITLE
 why detaching the state values when computing the advantage functions

### ISSUE THREAD

**Issue by jingxixu** [nikhilbarhate99/PPO-PyTorch#43]
I am not sure why you detach the state values when computing the advantage functions? Specifically, I am talking about

```
advantages = rewards - state_values.detach()
```

Many thanks!

**Comment by nikhilbarhate99** [comment]
refer to #29
